src/protocol/util/websocket_server.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024 CasparCG (www.casparcg.com).
+ *
+ * This file is part of CasparCG (www.casparcg.com).
+ *
+ * CasparCG is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * CasparCG is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with CasparCG. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "protocol_strategy.h"
+
+#include <common/memory.h>
+#include <core/monitor/monitor.h>
+
+#include <boost/asio.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/websocket.hpp>
+
+#include <functional>
+#include <memory>
+#include <set>
+#include <string>
+
+namespace caspar { namespace IO {
+
+/**
+ * WebSocket server that handles both AMCP protocol commands and monitor data streaming.
+ */
+class websocket_server
+{
+  public:
+    // Constructor for AMCP + Monitor (legacy)
```

Comment:
a 'legacy' and a 'new' in some new code? sounds like one of these can be removed